### PR TITLE
.gitignore: Adding .idea to .gitignore

### PR DESCRIPTION
**Pull request by Rushit**
This will keep  all intellij IDEA IDE related files out of git.
This helps contributors using IDEA IDE for development.

/cc @mitake
as per https://github.com/coreos/etcd/pull/7229/files/2fe7945a6b3a5f099878c70aabb8c1ce24c5518f

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@
 *.test
 tools/functional-tester/docker/bin
 hack/tls-setup/certs
+.idea
```
Comment (heyitsanthony) [inline .gitignore]: include a newline?

Comment (Rushit) [inline .gitignore]: done.

**Comment by xiang90** [comment]
Please squash two commits into one. I will get this merged after the change.

**Comment by Rushit** [comment]
@xiang90 just squashed both commit to one.

**Pull request merged**